wandb/sdk/internal/internal_api.py

```diff
@@ -1942,18 +1946,79 @@ def upload_urls(
             project (str): The project to download
             files (list or dict): The filenames to upload
             run (str, optional): The run to upload to
-            entity (str, optional): The entity to scope this project to.  Defaults to wandb models
+            entity (str, optional): The entity to scope this project to.
             description (str, optional): description
 
         Returns:
-            (bucket_id, file_info)
-            bucket_id: id of bucket we uploaded to
-            file_info: A dict of filenames and urls, also indicates if this revision already has uploaded files.
+            (run_id, upload_headers, file_info)
+            run_id: id of run we uploaded files to
+            upload_headers: A list of headers to use when uploading files.
+            file_info: A dict of filenames and urls.
                 {
-                    'weights.h5': { "url": "https://weights.url" },
-                    'model.json': { "url": "https://model.json", "updatedAt": '2013-04-26T22:22:23.832Z', 'md5': 'mZFLkyvTelC5g8XnyQrpOw==' },
+                    "run_id": "run_id",
+                    "upload_headers": [""],
+                    "file_info":  [
+                        { "weights.h5": { "uploadUrl": "https://weights.url" } },
+                        { "model.json": { "uploadUrl": "https://model.json" } }
+                    ]
                 }
         """
+        run_name = run or self.current_run_id
+        assert run_name, "run must be specified"
+        entity = entity or self.settings("entity")
+        assert entity, "entity must be specified"
+
+        has_create_run_files_mutation = self.create_run_files_introspection()
+        if not has_create_run_files_mutation:
+            return self.legacy_upload_urls(project, files, run, entity, description)
+
+        query = gql(
+            """
+        mutation CreateRunFiles($entity: String!, $project: String!, $run: String!, $files: [String!]!) {
+            createRunFiles(input: {entityName: $entity, projectName: $project, runName: $run, files: $files}) {
+                runID
+                uploadHeaders
+                files {
+                    name
+                    uploadUrl
+                }
+            }
+        }
+        """
+        )
+
+        query_result = self.gql(
+            query,
+            variable_values={
+                "project": project,
+                "run": run_name,
+                "entity": entity,
+                "files": [file for file in files],
+            },
+        )
+
+        result = query_result["createRunFiles"]
+        run_id = result["runID"]
```

Comment:
should we error handle here? 
in case the response missing any of the fields or malformed?